[docs] Enterprise Landing Page (#5804)
hashicorp/consul: website/source/docs/enterprise/index.html.md

Licensing capabilities were added to Consul Enterprise v1.1.0. The license is set
once for a datacenter and will automatically propagate to all nodes within the
datacenter over a period of time scaled between 1 and 20 minutes depending on the
number of nodes in the datacenter. There are two methods for licensing Consul
enterprise.

### Included in the Enterprise Package

> If you are downloading Consul from Amazon S3, then the license is included

Reviewer: The temporary license binaries are available in S3 too. Up until the last month or so that was the only place any of the binaries existed.

The zip packages containing `+pro` or `+prem` in the name in S3 are the ones with builtin licenses. The zip packages that have `+ent` in the name are the same ones that we now copy to releases.hashicorp.com

Author: @mkeeler Thanks for the info! I've update the page, lmk if it needs additional clarification.

and you do not need to take further action. This is the most common use 
case.

### Applied after Bootstrapping

> If you are downloading the enterprise binary from the [public hub](https://releases.hashicorp.com/consul/), you will need to apply

Reviewer 2: ```suggestion
If you are downloading the enterprise binary from the [releases.hashicorp.com](https://releases.hashicorp.com/consul/), you will need to apply
```

> the license to the leading server after bootstrapping the cluster. 

Reviewer: You can set the license on any enterprise agent connected in the cluster and the request gets forwarded to the leader to be applied. So you can send it to either a client, non-leader server or the leader and it will work the same. Maybe we want to simplify things and just say to do it against the leader. I am fine either way.

> The license can be set via the 

Reviewer 3: ```suggestion
You can set the license via the 
```

[API](/api/operator/license.html) or the [CLI](/docs/commands/license.html). When

> Consul is first started, a 30 minute temporary license is available to allow for

Reviewer 3: ```suggestion
you first start Consul, a 30-minute temporary license is available to allow you
```

> time to license the datacenter. The license should be set within ten minutes of

Reviewer 3: ```suggestion
time to license the datacenter. You should set the license within ten minutes of
```
